src/sentry/api/endpoints/group_details.py

```diff
@@ -152,16 +152,27 @@ def _get_context_plugins(self, request, group):
             PluginSerializer(project),
         )
 
-    def _get_release_info(self, request, group, version):
+    def _get_release_object(self, request, group, version):
         try:
-            release = Release.objects.get(
+            return Release.objects.get(
                 projects=group.project,
                 organization_id=group.project.organization_id,
                 version=version,
             )
         except Release.DoesNotExist:
             return {"version": version}
-        return serialize(release, request.user)
+
+    def _get_release_info(self, request, group, version):
+        return serialize(self._get_release_object(request, group, version), request.user)
+
+    def _get_first_last_release_info(self, request, group, versions):
+        releases = [self._get_release_object(request, group, version) for version in versions]
```

Comment:
Would it be useful to make a query to fetch all the releases at once? Right now you're doing n-queries.